bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,335 @@
+                if (isVideoMute || isAudioMute) {
+                    if (command == RefreshType.REFRESH) {
+                        // refresh both video and audio mute, as it's one request
+                        MuteQueryResponseValue muteStatus = device.getMuteStatus();
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                                muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                        updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                                muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                    } else {
+                        if (isAudioMute) {
+                            logger.trace("Received audio mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                        }
+                        if (isVideoMute) {
+                            logger.trace("Received video mute command" + command);
+                            boolean muteOn = command == OnOffType.ON;
+                            device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                        }
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+    }
+
+
+    @Override
+    public void initialize() {
+        this.setupJob = scheduler.schedule(() -> {
+            try {
+                setupDevice();
+                setupRefreshInterval();
+            } catch (ResponseException | IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (AuthenticationException e) {
+                this.handleAuthenticationException(e);
+            } catch (ConfigurationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        }, 0, TimeUnit.SECONDS);
+    }
+
+    protected PJLinkDeviceConfiguration getConfiguration() throws ConfigurationException {
+      PJLinkDeviceConfiguration config = this.config;
+      try {
+          this.validateConfigurationParameters(this.getThing().getConfiguration().getProperties());
```

Comment:
I'm not sure if I understand what your trying to do here. The framework it self already validates the configuration parameters the user set values. So unless you set some configuration values (that are also defined in the thing xml, because that is what is validated against) it make sense to validate here. Otherwise I don't think this is needed.

Reply:
I noticed that parameters of things defined in .things files did not get validated, that's why I'm calling it manually here. This way the user will get information about parameter validation issues as part of the thing state details.

But I was also confused myself that this was not done by the framework, as it seems like a very common requirement. Maybe I'm doing something wrong, do you have an idea why parameters coming from .things files are not validated?

Reply:
Maybe they're validated but I think that a binding can be started with a parameter unset while this parameter is declared as required. Generally I add a control in initialize method. Maybe that is useless !  That has to be checked.

Reply:
And even if the validation is run by the framework, does it lock the binding intialization ? To be checked too.